### PR TITLE
feat: add sentry-sdk ^4.0 support

### LINKED ISSUE #140: SentryMiddleware incompatibility with sentry ^4.0
Hello guys!
Sentry team had changed signature of "flush" method from PromiseInterface:
https://github.com/getsentry/sentry-php/blob/3.22.1/src/ClientInterface.php#L81
to "Result"
https://github.com/getsentry/sentry-php/blob/4.0.0/src/ClientInterface.php#L78

And integration is seems to be broken and generates a lot of extra sentry events

```
Error: Call to undefined method Sentry\Transport\Result::wait()
  File "/public/index.php", line 7
    require_once dirname(__DIR__) . '/vendor/autoload_runtime.php';
...
(3 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by WaylandAce**
Closes https://github.com/Baldinof/roadrunner-bundle/issues/140

### CODE REVIEW

File: src/Integration/Sentry/SentryMiddleware.php [hunk 1]
```diff
@@ -26,7 +26,11 @@ public function process(Request $request, HttpKernelInterface $next): \Iterator
             yield $next->handle($request);
         } finally {
             $client = $this->hub->getClient();
-            $client?->flush()->wait(false);
+
+            $result = $client?->flush();
+            if (is_a($result, 'GuzzleHttp\Promise\PromiseInterface')) {
```
Comment (Baldinof) [inline src/Integration/Sentry/SentryMiddleware.php]: I would prefer `instanceof`

```suggestion
            if ($result instanceof PromiseInterface) {
```

Comment (WaylandAce) [reply src/Integration/Sentry/SentryMiddleware.php]: But there will be no PromiseInterface if guzzle is not installed.

Comment (Baldinof) [reply src/Integration/Sentry/SentryMiddleware.php]: It's not an issue with PHP, it will simply return `false`. See https://3v4l.org/p288k

**Comment by bitgandtter** [comment]
@WaylandAce i need this update as well. i can help if you give me access to your fork/repo

**Comment by Baldinof** [comment]
@bitgandtter  If you want to do another PR that use `instanceof` I will accept it

**Comment by WaylandAce** [comment]
Applied suggestion.

**Comment by Baldinof** [comment]
It needs the `use` too

**Comment by bitgandtter** [comment]
@WaylandAce that needs to be applied to the listener as well

```
    public function onWorkerStop(WorkerStopEvent $event): void
    {
        $result = $this->hub->getClient()?->flush();
        if ($result instanceof GuzzleHttp\Promise\PromiseInterface) {
            $result->wait(false);
        }
    }
```

**Comment by bitgandtter** [comment]
@Baldinof can you review?

**Review by bitgandtter — APPROVED** [review]

File: src/Integration/Sentry/SentryListener.php [hunk 2]
```diff
@@ -17,9 +17,10 @@ public function __construct(private HubInterface $hub)
 
     public function onWorkerStop(WorkerStopEvent $event): void
     {
-        $client = $this->hub->getClient();
-
-        $client?->flush()->wait(false);
+        $result = $this->hub->getClient()?->flush();
+        if ($result instanceof GuzzleHttp\Promise\PromiseInterface) {
```
Comment (Baldinof) [inline src/Integration/Sentry/SentryListener.php]:
```suggestion
        if ($result instanceof \GuzzleHttp\Promise\PromiseInterface) {
```

File: src/Integration/Sentry/SentryMiddleware.php [hunk 3]
```diff
@@ -25,8 +25,10 @@ public function process(Request $request, HttpKernelInterface $next): \Iterator
         try {
             yield $next->handle($request);
         } finally {
-            $client = $this->hub->getClient();
-            $client?->flush()->wait(false);
+            $result = $this->hub->getClient()?->flush();
+            if ($result instanceof GuzzleHttp\Promise\PromiseInterface) {
```
Comment (Baldinof) [inline src/Integration/Sentry/SentryMiddleware.php]:
```suggestion
            if ($result instanceof \GuzzleHttp\Promise\PromiseInterface) {
```

**Review by Baldinof — CHANGES_REQUESTED** [review]

**Review by Baldinof — APPROVED** [review]

**Comment by Baldinof** [comment]
It's released as `3.1.1`.

Thank you for the PR @WaylandAce

**Pull request merged**